internal/guest/storage/scsi/scsi.go

```diff
@@ -357,3 +415,60 @@ func UnplugDevice(ctx context.Context, controller, lun uint8) (err error) {
 	}
 	return nil
 }
+
+// runBlkidCommand execs a call to `blkid`. This function is broken out
+// so that we can stub it for unit testing of this package.
+func runBlkidCommand(args []string) (string, error) {
+	cmd := exec.Command("blkid", args...)
+	raw, err := cmd.Output()
+	return string(raw), err
+}
+
+// getDeviceFsType queries to find the filesystem on a dev device.
+// The mount(8) man page states that it uses `blkid` to determine
+// what filesystem a given device is using, so we do the same here.
+//
+// If a device is present but has no filesystem configured, which would be the
+// case for empty scratch disks for example, they will not show up in the
+// response from `blkid`, and `blkid` will return an error. Since `blkid`
+// returns the same error both for if a device is not present or a device
+// does not have a filesystem configured on it, getDeviceFsType should only
+// be called after we've validated that the device exists.
+//
+// `devicePath` is the /dev path to the target device.
+func getDeviceFsType(devicePath string) (string, error) {
+	rawOutput, err := _runBlkidCommand([]string{devicePath})
+	if err != nil {
+		// According to the blkid man page, blkid will return 2 if the specified token
```

Comment:
I don't think this is valid for BusyBox blkid. Based on a quick test (`touch foo; blkid foo`), it prints no info and exits with code 0.

Reply:
It may be fine to leave this in, though. It doesn't seem like it would hurt anything. Just a reminder that the version of tools we ship doesn't necessarily align with what is found in mainline distros.

Reply:
Good catch. I think this will need to change, since otherwise we wouldn't return from this call successfully when the device has no filesystem.

Reply:
Updated this. We return an empty string and no error if the response from `blkid` is empty and the error code was 0.